prow/config/tide.go

```diff
@@ -246,16 +306,52 @@ func (t *Tide) BatchSizeLimit(repo OrgRepo) int {
 // MergeMethod returns the merge method to use for a repo. The default of merge is
 // returned when not overridden.
 func (t *Tide) MergeMethod(repo OrgRepo) types.PullRequestMergeType {
-	v, ok := t.MergeType[repo.String()]
-	if !ok {
-		if ov, found := t.MergeType[repo.Org]; found {
-			return ov
-		}
-
-		return types.MergeMerge
+	mt, err := t.OrgRepoBranchMergeMethod(repo, "")
+	if err != nil {
+		// TODO(danilo-gemoli): return an error or validate regex beforehand
+		panic(err)
 	}
+	return mt
+}
 
-	return v
+func (t *Tide) OrgRepoBranchMergeMethod(repo OrgRepo, branch string) (types.PullRequestMergeType, error) {
+	orgRepoBranchShorthand := fmt.Sprintf("%s/%s", repo.String(), branch)
+	if orgRepoBranch, found := t.MergeType[orgRepoBranchShorthand]; found && orgRepoBranch.MergeType != "" {
+		return orgRepoBranch.MergeType, nil
+	}
+	if orgRepo, found := t.MergeType[repo.String()]; found && orgRepo.MergeType != "" {
+		return orgRepo.MergeType, nil
+	}
+	if org, found := t.MergeType[repo.Org]; found && org.MergeType != "" {
+		return org.MergeType, nil
+	}
+	mergeMethod := types.MergeMerge
+	for org, orgConfig := range t.MergeType {
+		// Match org or skip to the next
+		if matched, err := regexp.MatchString(org, repo.Org); err != nil {
```

Comment:
I'd say that all regexes should be pre-compiled (and also, validated by the respective tools) so that we can use simple `func (*Regexp) MatchString` that cannot error